src/mlpack/tests/main_tests/preprocess_binarize_test.cpp

```diff
@@ -0,0 +1,110 @@
+/**
+ * @file preprocess_binarize_test.cpp
+ * @author Manish Kumar
+ *
+ * Test mlpackMain() of preprocess_binarize_main.cpp.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include <string>
+
+#define BINDING_TYPE BINDING_TYPE_TEST
+static const std::string testName = "PreprocessBinarize";
```

Comment:
You can define `testName` after including `mlpack/core.hpp`, which will obviate the need for including `<string>` at the top.  (I know, it is a minor comment...)